lalrpop-test/src/lib.rs

```diff
@@ -11,104 +11,104 @@ use util::tok::Tok;
 
 /// Tests that actions can return the grammar's type parameters' associated
 /// types.
-mod associated_types;
+lalrpop_mod!(associated_types);
```

Comment:
Something to consider with the parser modules of `lalrpop-test` is that one may want to inspect the code of these. So keeping the genereated files in the same directory might be nice. On the other hand it is nice to remove all these entries from the .gitignore as well. Not sure which way is the best here.

Reply:
> Something to consider with the parser modules of lalrpop-test is that one may want to inspect the code of these. So keeping the genereated files in the same directory might be nice.

I think that a dedicated repository (like https://github.com/lalrpop/samples) is better for that purpose. Thus, (a) people don't need to download and compile the repository in order to see the generated code, and (b) it can be read directly in GitHub.

I created a [lalrpop-samples](https://github.com/ayosec/lalrpop-samples) repository as an example.

> On the other hand it is nice to remove all these entries from the .gitignore as well. Not sure which way is the best here.

I didn't see that. I updated the `.gitignore` file in 2c9b14487cda6e203a02366f7020d91006e764bf.

Reply:
> I think that a dedicated repository (like https://github.com/lalrpop/samples) is better for that purpose. Thus, (a) people don't need to download and compile the repository in order to see the generated code, and (b) it can be read directly in GitHub.

I were thinking of this in terms of developing on lalrpop and changing the codegen. In that case it is easier to look at the generated code if it is directly in the repo structure instead of inside `target/debug/deps/lalrpop-712387hgsagd/...`. I will let @nikomatsakis have the final say on this though (I believe he is at mozilla all hands this week so he probably don't have time this week).

> I didn't see that. I updated the .gitignore file in 2c9b144.

:+1:

Reply:
On balance, not having to add to the gitignore file seems good. I was wondering though: when you are using the OUT_DIR, what happens if you get a compilation error? (I'll have to experiment myself). How good are we at giving you source error information?